### ISSUE TITLE
EmptyStatement: false negative on top-level empty declaration in compact source file

### ISSUE THREAD

**Issue by MichaHub** [checkstyle/checkstyle#20749]
Found during compact source coverage work for #20590 (PR #20725).

I have executed the cli and showed it below.
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyStatement"/>
  </module>
</module>
```
Compact source file (JEP 512):
```
$ cat Compact.java
;

void main() {
    ;

    boolean cond = true;
    for (; cond;);

    if (cond);

    while (cond) {
        cond = false;
        ;
    }
}
```

Equivalent ordinary source for comparison:
```
$ cat Ordinary.java
public class Ordinary {
    ;

    void main() {
        ;

        boolean cond = true;
        for (; cond;);

        if (cond);

        while (cond) {
            cond = false;
            ;
        }
    }
}
```
```
$ java -Duser.language=en -jar checkstyle-13.0.1-SNAPSHOT-all.jar -c config.xml Compact.java
$ java -Duser.language=en -jar checkstyle-13.0.1-SNAPSHOT-all.jar -c config.xml Ordinary.java
```
Starting audit...
[ERROR] C:\....\cmp-emptystatement\Compact.java:4:5: Empty statement. [EmptyStatement]
[ERROR] C:\....\cmp-emptystatement\Compact.java:7:18: Empty statement. [EmptyStatement]
[ERROR] C:\....\cmp-emptystatement\Compact.java:9:14: Empty statement. [EmptyStatement]
[ERROR] C:\....\cmp-emptystatement\Compact.java:13:9: Empty statement. [EmptyStatement]
Audit done.
Checkstyle ends with 4 errors.

Starting audit...
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:5:9: Empty statement. [EmptyStatement]
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:8:22: Empty statement. [EmptyStatement]
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:10:18:Empty statement. [EmptyStatement]
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:14:13:Empty statement. [EmptyStatement]
Audit done.
Checkstyle ends with 4 errors.

In both cases only the `;` inside the method body is reported. The
top-level `;` (compact) and the `;` between class members (ordinary)
produce no `EMPTY_STAT` token, so the check does not react to them.
The behavior of compact and ordinary source is consistent with each
other — this issue is about the compact case specifically.

**Describe what you expect in detail.**

A violation should be reported for the top-level `;` at line 1 of the
compact source file. This empty declaration outside of any method is
accepted by the compiler but serves no purpose, so EmptyStatement
should flag it.

The input file in PR #20725 marks this line with `// ok until <this issue>`.

**Comment by Carbon14-48** [comment]
On this
